opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -331,6 +340,22 @@ enum DataPointFlags {
   // Bits 2-31 are reserved for future use.
 }
 
+// GaugeSubType defines valid gauge "sub-types", which are used to describe
+// "special" kinds of gauges which have additional properties. These sub-types
+// generally apply to metrics which originate from protocols which support
+// these additional types, but are represented as gauges in OTLP.
+enum GaugeSubType {
+  // GAUGE_SUBTYPE_NONE is the default GaugeSubType, which represents a gauge
+  // that does not have a sub_type.
+  GAUGE_SUBTYPE_NONE = 0;
+
+  // GAUGE_SUBTYPE_UNKNOWN is a GaugeSubType which represents a metric for
+  // which the type is not known, such as a Prometheus metric without a TYPE
+  // comment. For example, the metric may be a monotonically-increasing counter,
+  // or histogram, but type information was not included at the source.
+  GAUGE_SUBTYPE_UNKNOWN = 1;
```

Comment:
```suggestion
  GAUGE_SUBTYPE_PROMETHEUS_UNKNOWN = 1;
```
Would it make sense to indicate that this is not part of the OTel spec/datamodel/API but is specific only to Prometheus?

Reply:
I would prefer to keep prometheus as an example of something that _could_ use the enum, but not make the enum value prometheus-specific.  I think OTLP should avoid fields that are specific to another protocol, even if we don't currently have plans to use them for other purposes.  For example, I wouldn't have wanted the Summary type to be called PrometheusSummary, even if the initial usage is only for Prometheus.  I'm no expert, but there are other "typeless" protocols, like carbon, which might want to use this field.

I agree it makes sense to clarify that this cannot be produced by the OpenTelemetry API.  I'll make the update.

I'm not sure what it would mean to say it isn't part of the OTel datamodel, but is part of OTLP, since OTLP [is part of](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#opentelemetry-protocol-data-model) the OTel datamodel.

Reply:
GAUGE_SUBTYPE_NONE now includes: `Since the OpenTelemetry API enforces types, this subtype is used for all metrics from OpenTelemetry instrumentation.`

Reply:
For data coming from the OpenTelemetry API, I would not expect any untyped/unknown data to be allowed there even in the future. If there is even just a slim chance that it might be used for other use cases, however, then keeping the enum values neutral but adding a comment should be fine, yes. The clarification in 24a76be LGTM.